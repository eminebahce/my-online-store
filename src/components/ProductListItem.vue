<template>
  <v-card :loading="loading" class="pa-2" max-width="380" tile>
    <template slot="progress">
      <v-progress-linear
        color="deep-purple"
        height="10"
        indeterminate
      ></v-progress-linear>
    </template>

    <v-img height="250" :src="product.modelList[0].thumbUrl"></v-img>

    <v-card-title>{{ product.fmyMarketingName }}</v-card-title>

    <v-card-text>
      <v-row align="center" class="mx-0">
        <v-rating
          :value="product.modelList[0].ratings"
          color="amber"
          dense
          half-increments
          readonly
          size="14"
        ></v-rating>

        <div class="grey--text ms-4">
          {{ parseFloat(product.modelList[0].ratings) }} ({{
            product.modelList[0].reviewCount
          }})
        </div>
      </v-row>

      <div class="my-4 text-subtitle-1">
        $ {{ product.modelList[0].priceDisplay }} â€¢
        {{ product.modelList[0].pviTypeName }},
        {{ product.modelList[0].pviSubtypeName }}
      </div>

      <div>
        {{ product.modelList[0].usp[0] }}, {{ product.modelList[0].usp[1] }},
        {{ product.modelList[0].usp[2] }}
      </div>
    </v-card-text>
    <v-card-actions>
      <v-row align="center" justify="space-around">
        <v-btn
          color="deep-purple accent-4"
          small
          dark
          class="mb-4 mt-4"
          @click="addToCart(product)"
        >
          Add to cart
          <v-icon right>mdi-basket</v-icon>
        </v-btn>
      </v-row>
    </v-card-actions>
  </v-card>
</template>

<script>
export default {
  name: "ProductListItem",
  props: ["product"],

  methods: {
    addToCart(product) {
      this.$router.push({
        name: "OrderOverview",
        params: { product: product },
      });
    },
  },
};
</script>

<style scoped>
</style>