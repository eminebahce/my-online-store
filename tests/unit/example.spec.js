import { createLocalVue, mount } from '@vue/test-utils'
import Vuex from 'vuex';
import ProductListItem from '../../src/components/ProductListItem.vue'

const localVue = createLocalVue();
localVue.use(Vuex);

let actions;
let store;
let mutations;

const data = {
  allProducts: [],
};


describe('ProductList.vue', () => {

  beforeEach(() => {
    actions = {
        "getAllProducts": jest.fn(() => Promise.resolve()),
    };
    store = new Vuex.Store({
        actions,
        mutations,
    });
    mutations = {
        "GET_ALL_PRODUCTS": jest.fn()
    }
});

test('GET_ALL_PRODUCTS', async () => {
  const state = {
    allProducts: [],
  };      
  expect(state.allProducts).toStrictEqual(data.allProducts);
});
})

describe('ProductList component', () => {
 test('renders aProductListItem component with all products props', () => {
  var product = {
    familyRecord: "1",
    familyId: "344232",
    modelCount: "8",
    fmyMarketingName: "Galaxy Tab S7 (11.0\", Wi-Fi)",
    fmyEngName: "Galaxy Tab S7 11.0 Wi-Fi",
    categorySubTypeCode: "01020100",
    categorySubTypeEngName: "Galaxy Tab S",
    simplePdYN: "Y",
    oldProductYN: "N",
    productGroupId: "galaxytabs7t870",
    optionPriceUse: null,
    iaCtaDisplay: "Y",
    wtbOnlineDispFuncUseYN: null,
    chipOptions: [],
    awardList: [],
    quickLooks: [],
    modelList: [
      {
        thumbUrl: '//images.samsung.com/is/image/samsung/p6pim/nl/sm-t870ndbaeub/gallery/nl-galaxy-tab-s7-t870-344232-sm-t870ndbaeub-thumb-388780700',
        ratings: '4.87',
        priceDisplay: '699',
        pviTypeName: 'Modal',
        pviSubtypeName: 'Tablet',
        usp: [
          "11 inch display",
          "120 Hz",
          "Inclusief geavanceerde S Pen"
        ]
      }
    ]
    }

  const wrapper = mount(ProductListItem, {
    propsData: {
      product: product
    }
  })

  expect(wrapper.props().product.fmyMarketingName).toMatch("Galaxy Tab S7 (11.0\", Wi-Fi)");
  expect(wrapper.props().product.modelList[0].priceDisplay).toMatch("699");
  expect(wrapper.props().product.modelList[0].ratings).toMatch("4.87");
  expect(wrapper.props().product.modelList[0].thumbUrl).toMatch("//images.samsung.com/is/image/samsung/p6pim/nl/sm-t870ndbaeub/gallery/nl-galaxy-tab-s7-t870-344232-sm-t870ndbaeub-thumb-388780700");
  expect(wrapper.props().product.modelList[0].pviTypeName).toMatch("Modal");
  expect(wrapper.props().product.modelList[0].pviSubtypeName).toMatch("Tablet");
  expect(wrapper.props().product.modelList[0].usp[0]).toMatch("11 inch display");
  expect(wrapper.props().product.modelList[0].usp[1]).toMatch("120 Hz");
  expect(wrapper.props().product.modelList[0].usp[2]).toMatch("Inclusief geavanceerde S Pen");
});
})

describe("Snapshot tests", () => {
  test("ProductListItem Component", () => {
    var product = {
      familyRecord: "1",
      familyId: "344232",
      modelCount: "8",
      fmyMarketingName: "Galaxy Tab S7 (11.0\", Wi-Fi)",
      fmyEngName: "Galaxy Tab S7 11.0 Wi-Fi",
      categorySubTypeCode: "01020100",
      categorySubTypeEngName: "Galaxy Tab S",
      simplePdYN: "Y",
      oldProductYN: "N",
      productGroupId: "galaxytabs7t870",
      optionPriceUse: null,
      iaCtaDisplay: "Y",
      wtbOnlineDispFuncUseYN: null,
      chipOptions: [],
      awardList: [],
      quickLooks: [],
      modelList: [
        {
          thumbUrl: '//images.samsung.com/is/image/samsung/p6pim/nl/sm-t870ndbaeub/gallery/nl-galaxy-tab-s7-t870-344232-sm-t870ndbaeub-thumb-388780700',
          ratings: '4.87',
          priceDisplay: '699',
          pviTypeName: 'Modal',
          pviSubtypeName: 'Tablet',
          usp: [
            "11 inch display",
            "120 Hz",
            "Inclusief geavanceerde S Pen"
          ]
        }
      ]
      }

    const wrapper = mount(ProductListItem,{
      propsData: {
        product: product
      }
    })

    expect(wrapper.html).toMatchSnapshot()
  })
})
