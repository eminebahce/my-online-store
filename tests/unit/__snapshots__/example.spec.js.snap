// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests ProductListItem Component 1`] = `[Function]`;
